packages/plugins/src/PluginsContainer.ts

```diff
@@ -76,11 +95,11 @@ export class PluginsContainer {
                 `There is a requirement for plugin of type "${type}" to be only one registered.`
             );
         }
-        return list[0];
+        return this.decorate(list[0]) as T;
```

Comment:
Unnecessary this.decorate as it is already done in `atLeastOneByType -> byType -> findByType` call.